simplemma/language_detector.py

```diff
@@ -22,10 +22,13 @@ def __init__(self, tokenizer: Tokenizer = Tokenizer(SPLIT_INPUT)) -> None:
     def sample_tokens(self, text: str) -> List[str]:
         """Extract potential words, scramble them, extract the most frequent,
         some of the rest, and return at most 1000 tokens."""
-        # generator expression to split the text
-        counter = Counter(
-            token for token in self.tokenizer.split_text(text) if not token[0].isupper()
-        )
+
+        counter = Counter(token for token in self.tokenizer.split_text(text))
+
+        deletions = [token for token in counter if token[0].isupper()]
+        if len(deletions) < 0.8 * len(counter):
```

Comment:
I think that this should be configurable by the user (through the constructor) and documented in the docs :)